chia/full_node/mempool.py

```diff
@@ -1,32 +1,30 @@
 from __future__ import annotations
 
+import dataclasses
 import logging
-from datetime import datetime
 from typing import Dict, List, Optional
 
 from sortedcontainers import SortedDict
 
-from chia.full_node.bitcoin_fee_estimator import create_bitcoin_fee_estimator
 from chia.full_node.fee_estimation import FeeMempoolInfo
 from chia.full_node.fee_estimator_interface import FeeEstimatorInterface
 from chia.types.blockchain_format.coin import Coin
 from chia.types.blockchain_format.sized_bytes import bytes32
-from chia.types.clvm_cost import CLVMCost
-from chia.types.fee_rate import FeeRate
 from chia.types.mempool_item import MempoolItem
-from chia.util.ints import uint64
 
 
 class Mempool:
-    def __init__(self, max_size_in_cost: int, minimum_fee_per_cost_to_replace: uint64, max_block_cost_clvm: uint64):
+    def __init__(self, mempool_info: FeeMempoolInfo, fee_estimator: FeeEstimatorInterface):
```

Comment:
By the way, I'm not too jazzed about passing more info into the Mempool constructor than it needs, but as the Americans say "it is what it is".

Right now, we re-create the Mempool every block. The correct place for the calls into the fee estimator is behind `add_to_pool` and `remove_from_pool`, but the MempoolManager has to hold the fee_estimator ref to ensure it lives for the lifetime of the FullNode.

If we want to decouple more, we could pass some lambdas at Mempool construction time, or something else. I'm open to ideas!

Reply:
presumably the majority of blocks hit the "optimized" path though, where we just incrementally remove mempool items that were confirmed in the most recent block.

My understanding is that we only recreate the mempool when we experience a reorg

Reply:
Yes, that's right - we only run the other path during reorgs or at startup.